### PR TITLE
projects

### PR DESCRIPTION
**Pull request by chpy04**
## Changes

projects fast

## Notes

_Any other notes go here_

## Test Cases

- Case A
- Edge case
- ...

## Screenshots

_If you made UI changes you must post a screenshot of the whole page for each change in 1) a normal sized window and 2) the smallest possible window_

_If you did any manual testing (e.g., with Postman), put screenshots of the http request and before and after of the db_

_If none of this applies, you can delete this section_

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)

### CODE REVIEW

**Review by jr1221 — CHANGES_REQUESTED** [review]
I think file extensions should be `.rs`

File: src/backend/src/services/projects.services.ts [hunk 1]
```diff
@@ -100,12 +131,18 @@ export default class ProjectsService {
...
 
-    return projects.map(projectPreviewTransformer);
+    return projects.map(projectOverviewTransformer);
   }
 
+  /**
+   * Get the projects for a given team
+   * @param organization
+   * @param teamId
+   * @returns
```
Comment (walker-sean) [inline src/backend/src/services/projects.services.ts]: returns what???!?!?!??!?!?!?!??!?!?

File: src/frontend/src/apis/projects.api.ts [hunk 2]
```diff
@@ -12,18 +12,32 @@ import {
...
+  linkTypeTransformer,
+  projectOverviewTransformer,
+  projectGanttTransformer,
+  projectPreviewTransformer,
+  projectTransformer
+} from './transformers/projects.transformers';
 import { CreateSingleProjectPayload, EditSingleProjectPayload } from '../utils/types';
 
 /**
  * Fetches all projects.
```
Comment (walker-sean) [inline src/frontend/src/apis/projects.api.ts]: update the js docs plz

File: src/frontend/src/hooks/projects.hooks.ts [hunk 3]
```diff
@@ -32,30 +34,38 @@ import {
...
   deleteAbbreviation,
-  getTeamsProjects
+  getTeamsProjects,
+  getAllProjectPreviews
 } from '../apis/projects.api';
 import { CreateSingleProjectPayload, EditSingleProjectPayload } from '../utils/types';
 import { useCurrentUser } from './users.hooks';
 
 /**
  * Custom React Hook to supply all projects.
```
Comment (walker-sean) [inline src/frontend/src/hooks/projects.hooks.ts]: same here

File: src/frontend/src/hooks/projects.hooks.ts [hunk 4]
```diff
@@ -32,30 +34,38 @@ import {
...
-  return useQuery<ProjectPreview[], Error>(['projects'], async () => {
-    const { data } = await getAllProjects(includeDeleted);
+export const useAllProjectsGantt = () => {
+  return useQuery<ProjectGantt[], Error>(['projects'], async () => {
+    const { data } = await getAllProjectsGantt();
+    return data;
+  });
+};
+
+export const useAllProjectsPreviews = () => {
```
Comment (walker-sean) [inline src/frontend/src/hooks/projects.hooks.ts]: You could just call this `useAllProjects` and the type could just be `Project` if the old hook and type are bye bye

**Review by walker-sean — CHANGES_REQUESTED** [review]
🚀

**Comment by chpy04** [comment]
> I think file extensions should be `.rs`

@copilot fix pls
